libyul/AsmParser.cpp

```diff
@@ -158,22 +158,21 @@ Statement Parser::parseStatement()
 		}
 		while (currentToken() == Token::Comma);
 
-		expectToken(Token::Colon);
-		expectToken(Token::Assign);
+		expectToken(Token::AssemblyAssign);
 
 		assignment.value.reset(new Expression(parseExpression()));
 		assignment.location.end = locationOf(*assignment.value).end;
 		return Statement{std::move(assignment)};
 	}
+	case Token::AssemblyAssign:
 	case Token::Colon:
 	{
 		if (elementary.type() != typeid(Identifier))
 			fatalParserError("Label name / variable name must precede \":\".");
 		Identifier const& identifier = boost::get<Identifier>(elementary);
-		advance();
 		// identifier:=: should be parsed as identifier: =: (i.e. a label),
```

Comment:
This should be simplified now.

Reply:
I am not sure how. If we get `identifier:=:`, the scanner will make it `<identifier><AssemblyAssign> <Colon>` and if we get `identifier:` it will be `<identifier> <colon>`. If I want to handle both without repeating code I need to do it more or less  like this.

Reply:
Previously, spaces had to be ignored because we did not have a proper token for assembly assignment. After this change, `: =` and `:=` are two completely different things. So `Identifier:=:` (without spaces) should always be `Identifier:= :`, and only `Identifier: =:` should be a label (followed by a stack assignment).

Reply:
In other words, we don't need to allow/parse `Identifier:=:` anymore because it should be `Identifier: =:` anyway and that wasn't possible before

Reply:
Exactly. `Identifier:=:` was not explicitly meant "without spaces" anyway.